roles/tas_single_node/tasks/podman/restore.yml

```diff
@@ -0,0 +1,80 @@
+- name: Create /tmp/restore_mode_active
+  ansible.builtin.file:
+    path: /tmp/restore_mode_active
+    state: touch
+    mode: "0644"
+
+- name: Extract Basename from File
+  ansible.builtin.set_fact:
+    restore_file: "{{ tas_single_node_data_protection.restore.source | basename }}"
+
+- name: Push Encrypted Backup Tarball to Remote Server
+  ansible.builtin.copy:
+    src: "{{ tas_single_node_data_protection.restore.source }}.tar.gz.enc"
+    dest: "/root/{{ restore_file }}.tar.gz.enc"
+    remote_src: false
+    mode: "0600"
+
+- name: Decrypt Backup Tarball
+  ansible.builtin.shell: |
+    openssl enc -d -aes-256-cbc -salt -pbkdf2 -iter 600000 \
+    -in "{{ restore_file }}.tar.gz.enc" \
+    -out "{{ restore_file }}.tar.gz" \
+    -pass pass:"{{ tas_single_node_data_protection.restore.passphrase }}"
+  args:
+    executable: /bin/bash
+    chdir: /root/
+  register: tarball
+  changed_when: true
+
+- name: Delete Encrypted Backup Tarball on Remote Server
+  ansible.builtin.file:
+    path: "/root/{{ restore_file }}.tar.gz.enc"
+    state: absent
+
+- name: Extract Compressed Backup Tarball
+  ansible.builtin.unarchive:
+    src: "/root/{{ restore_file }}.tar.gz"
+    dest: /root/
+    remote_src: true
+
+- name: Delete Backup Tarball on Remote Server
+  ansible.builtin.file:
+    path: "/root/{{ restore_file }}.tar.gz"
+    state: absent
+
+- name: Stop Podman Containers
+  ansible.builtin.shell: podman stop $(podman ps -aq)
+  register: podman_stop_result
+  changed_when: podman_stop_result.rc == 0 and podman_stop_result.stdout | length > 0
+
+- name: Restore Podman Volumes
```

Comment:
Similar to my comment on the backup script, this will need to explicitly mount the podman volumes and copy their content to the mounted location.